ccm-connector/src/main/java/com/sequenceiq/cloudbreak/ccmimpl/cloudinit/DefaultCcmParameterSupplier.java

```diff
@@ -0,0 +1,118 @@
+package com.sequenceiq.cloudbreak.ccmimpl.cloudinit;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import com.cloudera.thunderhead.service.minasshdmanagement.MinaSshdManagementProto;
+import com.cloudera.thunderhead.service.minasshdmanagement.MinaSshdManagementProto.GenerateAndRegisterSshTunnelingKeyPairResponse;
+import com.cloudera.thunderhead.service.minasshdmanagement.MinaSshdManagementProto.MinaSshdService;
+import com.google.common.base.Throwables;
+import com.sequenceiq.cloudbreak.ccm.cloudinit.CcmParameterSupplier;
+import com.sequenceiq.cloudbreak.ccm.cloudinit.CcmParameters;
+import com.sequenceiq.cloudbreak.ccm.cloudinit.DefaultCcmParameters;
+import com.sequenceiq.cloudbreak.ccm.cloudinit.DefaultInstanceParameters;
+import com.sequenceiq.cloudbreak.ccm.cloudinit.DefaultServerParameters;
+import com.sequenceiq.cloudbreak.ccm.cloudinit.DefaultTunnelParameters;
+import com.sequenceiq.cloudbreak.ccm.endpoint.BaseServiceEndpoint;
+import com.sequenceiq.cloudbreak.ccm.endpoint.HostEndpoint;
+import com.sequenceiq.cloudbreak.ccm.endpoint.KnownServiceIdentifier;
+import com.sequenceiq.cloudbreak.ccmimpl.altus.GrpcMinaSshdManagementClient;
+
+/**
+ * Default supplier of CCM parameters, which:
+ * <ol>
+ * <li>contacts the minasshd management server via GRPC to acquire a minasshd instance;</li>
+ * <li>creates corresponding server parameters;</li>
+ * <li>contacts the minasshd instance via GRPC to register an instance and generate a keypair;</li>
+ * <li>creates the corresponding instance parameters; and</li>
+ * <li>creates the tunnel parameters based on the passed-in arguments.</li>
+ * </ol>
+ */
+@Component
+public class DefaultCcmParameterSupplier implements CcmParameterSupplier {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(DefaultCcmParameterSupplier.class);
+
+    @Inject
+    private GrpcMinaSshdManagementClient grpcMinaSshdManagementClient;
+
+    @Override
+    public Optional<CcmParameters> getBaseCcmParameters(@Nonnull String actorCrn, @Nonnull String accountId, @Nonnull String keyId) {
+        if (grpcMinaSshdManagementClient == null) {
+            return Optional.empty();
+        }
+        // JSA TODO get request ID from somewhere?
```

Comment:
I've been pulling it out of the MDC. I feel ok about this because it's only used for logging purposes and not for any of my own logic. I have seen some other modules use a ThreadBasedRequestIdProvider class

Reply:
ok